gremlin-go/driver/graph.go

```diff
@@ -120,3 +121,38 @@ func (p *Path) GetPathObject(key string) (interface{}, error) {
 		return nil, errors.New(fmt.Sprintf("Path does not contain a label of '%s'.", key))
 	}
 }
+
+// Set is a custom declaration since Go does not natively provide this feature.
+// Usage: Create a new Set from a slice using the createNewSet function.
+// createNewSet will remove all duplicate values from a slice and this new Set Object can then be serialized.
+type Set struct {
+	objects []interface{}
+}
```

Comment:
I think we should make the Set an interface and then have a struct that implements it with the underlying objects[]interface{}.

That way if a user wanted to, they could create their own set implementation based on our interface. Also should make objects -> Objects so that it is exprted and accessible.